File: others/2778-sum_of_squares.rb
# @param {Integer[]} nums
# @return {Integer}
def sum_of_squares(nums)
  result = 0
  for i in 0...(nums.length)
    if nums.length % (i + 1) == 0
      result += (nums[i] * nums[i])
    end
  end
  return result
end

File: others/2778-sumOfSquares.c
int sumOfSquares(int* nums, int numsSize){
  int i;
  int result = 0;
  for (i = 0; i < numsSize; i++) {
    if (numsSize % (i + 1) == 0) {
      result += (nums[i] * nums[i]);
    }
  }
  return result;
}
